chain/client/src/client.rs

```diff
@@ -810,19 +810,28 @@ impl Client {
             return NetworkClientResponses::InvalidTx(InvalidTxError::Expired);
         }
 
-        if self.runtime_adapter.cares_about_shard(me, &head.last_block_hash, shard_id, true) {
+        if self.runtime_adapter.cares_about_shard(me, &head.last_block_hash, shard_id, false)
```

Comment:
hm, I think it should is_me=true - because you might be tracking shard but not a validator for it - and then can still validate tx before routing it.